### PR TITLE
refactor: pallet improvements and xyk pallet

### PR DESCRIPTION
**Pull request by enthusiastmartin**
## Description
Amm pallet refactor to remove HDX references and use of native asset instead.

Added XYK pallet - basically amm pallet renamed and amm pallet will be eventually replaced by this new xyk pallet ( first it needs few clarifications) and amm deprecated. 

Few other minor code improvements as well.

## Motivation and Context
Some pallets are used in basilisk node too where there is other than HDX asset as native currency. So it would make sense to remove hdx references from the code to be more generic.

## How Has This Been Tested?
Tests.

### CODE REVIEW

File: pallets/xyk/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks};
+use frame_system::RawOrigin;
+use sp_std::prelude::*;
+
+use crate::Module as AMM;
```
Comment (jak-pan) [inline pallets/xyk/src/benchmarking.rs:26]: module or pallet?

Comment (enthusiastmartin) [reply pallets/xyk/src/benchmarking.rs]: pallet, pallet. for some reason, i did not get the deprecation warning here, so missed it. 

anyway, fixed.

File: pallets/amm/src/lib.rs [hunk 2]
```diff
@@ -123,11 +123,12 @@ pub mod pallet {
...
 		InsufficientAssetBalance,
-		InsufficientPoolAssetBalance,      // No tests
-		InsufficientNativeCurrencyBalance, // No tests
+
+		/// Not enough asset liquidity in the pool.
+		InsufficientPoolAssetBalance, // No tests
 
 		/// Not enough core asset liquidity in the pool.
-		InsufficientHDXBalance, // No tests
+    InsufficientNativeCurrencyBalance, // No tests
```
Comment (jak-pan) [inline pallets/amm/src/lib.rs:131]: rustfmt?

File: pallets/amm/src/lib.rs [hunk 3]
```diff
@@ -656,7 +657,10 @@ impl<T: Config> AMM<T::AccountId, AssetId, AssetPair, Balance> for Pallet<T> {
 		Ok(transfer)
 	}
 
-	#[transactional]
+	/// Execute sell. validate_sell must be called first.
+	/// Perform necessary storage/state changes.
+	/// Note : the execution should not return error as everything was previously verified and validated.
+  #[transactional]
```
Comment (jak-pan) [inline pallets/amm/src/lib.rs:663]: rustfmt

File: pallets/amm/src/lib.rs [hunk 4]
```diff
@@ -783,7 +787,10 @@ impl<T: Config> AMM<T::AccountId, AssetId, AssetPair, Balance> for Pallet<T> {
 		Ok(transfer)
 	}
 
-	#[transactional]
+	/// Execute buy. validate_buy must be called first.
+	/// Perform necessary storage/state changes.
+	/// Note : the execution should not return error as everything was previously verified and validated.
+  #[transactional]
```
Comment (jak-pan) [inline pallets/amm/src/lib.rs:793]: rustfmt

**Comment by jak-pan** [comment]
Let's remove amm

**Comment by enthusiastmartin** [comment]
> Let's remove amm

ready. Should I test it in some specific way? obviously it is ok when starting a new chain.

**Review by lumir-mrkva — APPROVED** [review]

**Comment by lumir-mrkva** [comment]
@enthusiastmartin LGTM

**Pull request merged**